pkg/apiserver/handlers/featuregates/handler.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package featuregates
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/antctl/runtime"
+	"github.com/vmware-tanzu/antrea/pkg/controller/querier"
+	"github.com/vmware-tanzu/antrea/pkg/features"
+)
+
+var controllerOnlyGates = []string{
+	"Traceflow",
+	"AntreaPolicy",
+	"Egress",
+	"NetworkPolicyStats",
+}
+
+type (
+	Config struct {
+		// FeatureGates is a map of feature names to bools that enable or disable experimental features.
+		FeatureGates map[string]bool `yaml:"featureGates,omitempty"`
+	}
+
+	Response struct {
+		Component      string `json:"component,omitempty"`
+		FeatureName    string `json:"name,omitempty"`
+		FeatureStatus  string `json:"status,omitempty"`
+		FeatureVersion string `json:"version,omitempty"`
+	}
+)
+
+// HandleFunc returns the function which can handle queries issued by featuregates commands.
+// The handler function populates Antrea featuregates information to the response.
+func HandleFunc(cq querier.ControllerQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		k8sClientset := cq.GetK8sClient()
+		antreaConfigMaps, err := k8sClientset.CoreV1().ConfigMaps("kube-system").List(context.TODO(), metav1.ListOptions{LabelSelector: "app=antrea"})
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			klog.Errorf("Error when getting Antrea app config maps: %v", err)
+		}
+		antreaConfigName := ""
+		for _, v := range antreaConfigMaps.Items {
+			if strings.Contains(v.Name, "antrea-config") {
+				antreaConfigName = v.Name
+				break
+			}
+		}
```

Comment:
seems the configmap's name is dynamicly generated by kcustomize, I haven't found a valid way to get it and pass it to container as env, so plan to get pod spec and get the exact name since we have env.GetPodName().

Reply:
sent you some code on Slack to export the ConfigMap name generated by kustomize as an env variable

Reply:
@antoninbas thanks for the tips, I have refined it to get the configmap name via env.